cfme/cloud/security_group.py

```diff
@@ -1,20 +1,268 @@
+        Version.lowest(): NonJSSecurityGroupsEntity,
+        '5.9': JSBaseEntity,
+    })
+
+
+class SecurityGroupsEntities(BaseEntitiesView):
+    @property
+    def entity_class(self):
+        return SecurityGroupsEntity().pick(self.browser.product_version)
+
+
+class SecurityGroupsDetailsEntities(View):
+    breadcrumb = BreadCrumb()
+    title = Text('//div[@id="main-content"]//h1')
+    table = Table("//div[@id='list_grid']//table")
+    properties = SummaryTable(title='Properties')
+    relationships = SummaryTable(title='Relationships')
+    smart_management = SummaryTable(title='Smart Management')
+    firewall_rules = SummaryTable(title="Firewall Rules")
+
+
+class SecurityGroupsAddEntities(View):
+    breadcrumb = BreadCrumb()
+    title = Text('//div[@id="main-content"]//h1')
+    network_manager = BootstrapSelect(id='ems_id')
+    cloud_tenant = Dropdown(text='admin')
+
+
+class SecurityGroupsAddForm(View):
+    network_manager = BootstrapSelect(id='ems_id')
+    security_groups_name = TextInput(name='name')
+    security_groups_description = TextInput(name='description')
+    cloud_tenant = Select(name='cloud_tenant_id')
+    add = Button('Add')
+    cancel = Button('Cancel')
+
+
+class SecurityGroupsView(BaseLoggedInPage):
+    """Base view for header and nav checking, navigatable views should inherit this"""
+    @property
+    def in_security_groups(self):
+        return(
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ['Networks', 'Security Groups'] and
+            match_location(controller='security_group', title='Security Groups'))
+
+
+class SecurityGroupsAllView(SecurityGroupsView):
+    @property
+    def is_displayed(self):
+        return (
+            self.in_security_groups and
+            self.entities.title.text == 'Security Groups')
+
+    toolbar = View.nested(SecurityGroupsToolbar)
+    including_entities = View.include(SecurityGroupsEntities, use_parent=True)
+
+
+class SecurityGroupsDetailsView(SecurityGroupsView):
+    @property
+    def is_displayed(self):
+        expected_title = '{} (Summary)'.format(self.context['object'].name)
+        return (
+            self.in_security_groups and
+            self.entities.title.text == expected_title and
+            self.entities.breadcrumb.active_location == expected_title)
+
+    toolbar = View.nested(SecurityGroupsDetailsToolbar)
+    sidebar = View.nested(SecurityGroupsDetailsAccordion)
+    entities = View.nested(SecurityGroupsDetailsEntities)
+
+
+class SecurityGroupsAddView(SecurityGroupsView):
+    @property
+    def is_displayed(self):
+        return (
+            self.in_security_groups and
+            self.entities.breadcrumb.active_location == 'Add New Security Group' and
+            self.entities.title.text == 'Add New Security Group')
+
+    entities = View.nested(SecurityGroupsAddEntities)
+    form = View.nested(SecurityGroupsAddForm)
+
+
+class SecurityGroups(Navigatable):
```

Comment:
Please make this class name singular, as well as the var names in its functions - `security_groups_name` can just be `name`, etc.

Reply:
Changed.